### PR TITLE
[1403] Show banner when last degree deleted

### PR DESCRIPTION
**Pull request by JonF1**
### Context
https://trello.com/c/izwer1EG/1403-show-banner-when-last-degree-deleted
### Changes proposed in this pull request
* Creates new degrees confirm details controller which decides whether to render confirmation component or incomplete section component depending whether the trainee has a degree. 
* Add a degree text to button no longer exists, as now dealt with by incomplete section component
* Edit feature test to test that incomplete component renders in user journey
* Add view spec for personal details show page 
### Guidance to review
Delete a trainee's degree so that component renders like below 

Screenshot 2021-04-12 at 17 57 24
Screenshot 2021-04-12 at 17 58 25

### CODE REVIEW

**Comment by JonF1** [comment]
[Show banner when last degree deleted](https://trello.com/c/izwer1EG/1403-show-banner-when-last-degree-deleted)

File: app/components/trainees/confirmation/degrees/view.html.erb [hunk 1]
```diff
@@ -12,8 +12,12 @@
...
 
-<% if show_add_another_degree_button %>
+<% if show_add_another_degree_button && trainee_holds_one_or_more_degrees %>
   <div class="govuk-form-group">
     <%= render GovukButtonLinkTo::View.new(body: degree_button_text, url: trainee_degrees_new_type_path(trainee), class_option: "govuk-button--secondary") %>
   </div>
 <% end %>
+
+<% if degrees.empty? %>
+  <%= render IncompleteSection::View.new(title: t("components.incomplete_section.degree_details_not_provided"), link_text: t("components.incomplete_section.add_degree_details"), url: trainee_degrees_new_type_path(trainee), error: false) %>
```
[the remark this answers is not in the document]
Comment (JonF1) [reply app/components/trainees/confirmation/degrees/view.html.erb]: Makes complete sense! And thanks for going through it with me today, i've pushed up the changes. Also I'm not 100% on the view spec for personal details page that ive added

File: spec/views/trainees/personal_details/show.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe "trainees/personal_details/show.html.erb" do
+  context "with a trainee with no degree" do
+    let(:trainee) { create(:trainee) }
+
+    before do
```
Comment (AbigailMcP) [inline spec/views/trainees/personal_details/show.html.erb_spec.rb:9]: You have the same `before` block in your second context too so you can actually declare it just once outside of them both.

**Review by AbigailMcP — COMMENTED** [review]
Just one code clean up, then it's good for product review!

File: spec/views/trainees/personal_details/show.html.erb_spec.rb (hunk 2 above)
Comment (JonF1) [reply spec/views/trainees/personal_details/show.html.erb_spec.rb]: thanks, fixed!

**Review by AbigailMcP — APPROVED** [review]

**Comment by GeoffJohnson** [comment]
All good

**Pull request merged**